src/relayedTransactionV2Builder.ts

```diff
@@ -0,0 +1,81 @@
+import {Transaction} from "./transaction";
+import {TransactionPayload} from "./transactionPayload";
+import {AddressValue, BytesValue, ContractFunction, U64Value} from "./smartcontracts";
+import {IAddress, IChainID, IGasLimit, INonce, ITransactionPayload} from "./interface";
+import {ISigner} from "@elrondnetwork/erdjs-walletcore/out/interface";
+import {INetworkConfig} from "./interfaceOfNetwork";
+import {ErrGasLimitShouldBe0ForInnerTransaction, ErrInvalidRelayedV2BuilderArguments} from "./errors";
+
+export class RelayedTransactionV2Builder {
+
+    innerTransaction: Transaction | undefined;
+    innerTransactionGasLimit: IGasLimit | undefined;
+    netConfig: INetworkConfig | undefined;
+
+    /**
+     * Sets the inner transaction to be used. It has to be already signed and with gasLimit set to 0. These checks
+     * are performed on the build() method
+     *
+     * @param {Transaction} transaction The inner transaction to be used
+     */
+    setInnerTransaction(transaction: Transaction) : RelayedTransactionV2Builder {
+        this.innerTransaction = transaction;
+        return this;
+    }
+
+    /**
+     * Sets the gas limit to be used for the SC Call inside the inner transaction
+     *
+     * @param {IGasLimit} gasLimit The gas limit to be used. The inner transaction needs to have the gas limit set to 0,
+     * so this field will specify the gas to be used for the SC call of the inner transaction
+     */
+    setInnerTransactionGasLimit(gasLimit: IGasLimit) : RelayedTransactionV2Builder {
+        this.innerTransactionGasLimit = gasLimit;
+        return this;
+    }
+
+    /**
+     * Sets the network config to be used for building the relayed v2 transaction
+     *
+     * @param {INetworkConfig} netConfig The network configuration to be used
+     */
+    setNetworkConfig(netConfig: INetworkConfig) : RelayedTransactionV2Builder {
+        this.netConfig = netConfig;
+        return this;
+    }
+
+    /**
+     * Tries to build the relayed v2 transaction based on the previously set fields
+     *
+     * @throws ErrInvalidRelayedV2BuilderArguments
+     * @throws ErrGasLimitShouldBe0ForInnerTransaction
+     * @return Transaction
+     */
+    build() : Transaction {
+        if(!this.innerTransaction || !this.innerTransactionGasLimit || !this.netConfig || !this.innerTransaction.getSignature()) {
+            throw new ErrInvalidRelayedV2BuilderArguments();
+        }
+        if(this.innerTransaction.getGasLimit() !== 0){
```

Comment:
Comparison might fail if `getGasLimit` returns an object that implements `IGasLimit`. Perhaps allow type coercion to happen? `!=` / or add a test with both numbers and objects if this works as it is.